### DISCUSSION TITLE
Vercel Dev Not working With Svelte Routify

### DISCUSSION THREAD

**Discussion by alteredorange** [vercel/vercel#4584] [Help]
Hello! Big fan of what you guys are doing! everything works great, except for your Vercel Dev command. You can see what I mean by installing routify with `npx @sveltech/routify init`. I set the dev command to `npm run dev` so when I run `vercel dev` it starts the routify server correctly, but then it doesn't start the vercel dev server or do anything else.  Any help would be greatly appreciated, thanks!

**Comment by markozxuu** [comment]
The ports may be conflicting but I'm not entirely sure 🤔 

cc @styfle

**Reply by alteredorange** [reply]
I've tried it with the port flag as well, still doesn't help. Not sure what's causing the hang up. Everything works fine with netlify dev command.

**Comment by styfle** [comment]
I ran `npx @sveltech/routify init` and ended up with a `package.json` like this:

```js
{
  "name": "svelte-app",
  "version": "1.0.0",
  "scripts": {
    "dev": "run-p routify rollup",
    "dev-dynamic": "cross-env BUNDLING=dynamic npm run dev",
    "build": "npm run routify -- -b && rollup -c",
    "serve": "spassr --serve-spa --serve-ssr",
    "export": "routify export",
    "build:docker": "npm run build && ./scripts/docker/docker-build.sh",
    "deploy:now": "cd scripts/now && npm run deploy",
    "deploy:netlify": "cd scripts/netlify && npm run deploy",
    "rollup": "rollup -cw",
    "routify": "routify -D"
  },
  /*... trimmed for brevity */
}
```

Then I ran `vercel` which failed to deploy correctly because there is no `public` directory after `npm run build`.

Then I ran `vercel dev` which ran a dev server but it does not proxy serverless functions.

```
> spassr --serve-spa --serve-ssr

Serving spa on http://localhost:5000
Serving ssr on http://localhost:5005
``` 

The "Output Directory" will need to be overridden as well as the "Develop Command"  (perhaps something like `spassr --serve-spa --serve-ssr --port $PORT`)

**Reply by alteredorange** [reply]
Yes, setting the output directory to 'dist' in the vercel command line or in the online settings makes it work just fine for deployment.  The dev command seems to ignore the settings, and still looks for a public folder, no matter what settings you have.  But even copying the 'dist' folder to 'public', the vercel dev command still doesn't function, it just runs the npm dev command, and the vercel dev server never starts.  On the same project running the 'netlify dev' command works just fine, so I'm not sure where 'vercel dev' is getting hung up.

**Reply by styfle** [reply]
Is there a `--port` flag for the spassr CLI?

**Reply by alteredorange** [reply]
Not at the moment, what port would it need to be on?

**Reply by styfle** [reply]
The port changes each time but it is assigned the the `PORT` env var.

That's why the development command should be something like `spassr --serve-spa --serve-ssr --port $PORT`.

For example, if Gatsby is detected, we use `gatsby develop --port $PORT`.

**Reply by alteredorange** [reply]
Ah, yes it always defaults to 5000 for static and 5005 for SSR, but I'm not sure they're using environmental variables for that, I think it's hard coded.

**Comment by QGB** [comment]
How to custom project start cmd (not "build.sh") in vercel.json?
